### ISSUE TITLE
Configure pytest so one invocation runs all python tests?

### ISSUE THREAD

**Issue by mccalluc** [opendp/opendp#1022]
Right now, I think `smoke-test.yml` runs python tests in three different places:

Normal tests:
```
      - name: Test
        run: pytest -v
```

Doctests in docs:
```
      - name: Test docs
        run: |
          cd ../docs
          python -m pip install -r requirements.txt
          make doctest-python
```

Check that notebooks run without exceptions:
```
      - name: Test
        run: |
          shopt -s globstar
          pytest --nbmake source/**/*.ipynb -n=auto
```

It may be a little simpler to just run pytest once from the top level with appropriate configs to cover all of this. Replicating the doctests locally right now is slow, because they follow the full documentation build in `make doctest-python`. We might also be missing things: because we're running the tests in different places, we don't get one count for the total number of tests run... if we misconfigured something and a set of tests stopped running, we don't have a count or a coverage measure that could show that something changed.

**Comment by mccalluc** [comment]
I think this was closed inadvertently -- #1029 is related, but doesn't address this completely.

**Comment by Shoeboxam** [comment]
The notebooks are run in a separate task on purpose, so that both sets of tests can be executed simultaneously. It shaves up to ~3 minutes of time off of CI.
